giskard/rag/question_generators/ookb_questions.py

```diff
@@ -0,0 +1,184 @@
+import uuid
+
+from ..knowledge_base import KnowledgeBase
+from .base import _LLMBasedQuestionGenerator, GenerateFromSingleQuestionMixin
+from .prompt import QAGenerationPrompt
+
+OoKB_PROMPT = """
+You are a powerful auditor and mindful judger, your role is to generate question from a given context and 
+by adding some fake or non-existing details/facts to the context to check whether the agent model you are auditing is capable of reasoning and answering questions
+which has no direct answer in the provided context.
+
+There are your tasks, you should finish them step by step:
+0. select one detail/fact from the context
+1. add some fake details/facts which is not inclued in the whole provided context but based on the detail/facts you selected in the previous step.
+2. Isolate this added fake detail/fact into a single sentence
+3. Generate a question based on the new added isolated detail/fact, fit the question in the context and can not be answered by the information in the context.
+
+The user will provide the context and give your answer after read it thoroughly.
+You will return the isolated detail/fact and the question based exclusively on the new added isolated context.
+You must output a single JSON object with keys 'selected detail/fact', 'fake detail/fact' and 'question' , without any other wrapping text or markdown and everything is in low letter. Make sure you only return valid JSON. 
+"""
+
+OoKB_QUESTION_EXAMPLE_INPUT = """Paul Graham liked to buy a baguette every day at the local market."""
+
+OoKB_QUESTION_EXAMPLE_OUTPUT = """
+
+{   
+    "selected detail/fact": "Paul Graham liked to buy a baguette every day at the local market.",
+    "fake detail/fact": "Paul Graham paid 1 USD for a baguette",
+    "question": "How much did Paul pay for the baguette?"
+}
+
+"""
+
+QUESTION_CHECK_PROMPT = """
+You are a powerful auditor and mindful juder, your role is to check whether the input question can be answered by the context and the related context documents or not.
+
+The agent model you are auditing is the following:
+- Agent description: {agent_description}
+
+Your task is to check if the given question can be answered by the context and the related context or not. 
+In case the question is asking some facts in the context, unless the facts are explicitly mentioned in the context, the question can not be answered by the context.
+
+All the generated content should be in the following language: {language}
+
+Think step by step and consider the context in its entirety. Remember: you need to have a strong and sound reason to support your evaluation.
+If the question can be answered by the context, return True along with the reason. If the question can not be answered by the context, return False along with the reason.
+You must output a single JSON object with keys 'can_be_answered' and 'correctness_reason'. NO ANY MARKDOWN STRING. Make sure you return a valid JSON object.
+"""
```

Comment:
In the final form, we don't really need a reason or JSON format. We can just ask to get a single  token Y/N answer.

Reply:
But I am not sure without the reason, this could cause the problem of lazyness and reduce the accuracy of LLM's response.